### ISSUE TITLE
IOError: Permission denied on some files

### ISSUE THREAD

**Issue by awesomebytes** [dddomodossola/remi#37]
Hello, I was just trying your thing (looks pretty cool!) but I cannot run it correctly.

I just git cloned it then did:

    sudo python setup.py install

and entered the examples folder and executed the `widgets_overview_app.py` then I got:

```
127.0.0.1 - - [03/Nov/2015 21:25:15] "GET / HTTP/1.1" 200 -
127.0.0.1 - - [03/Nov/2015 21:25:15] "GET /res/style.css HTTP/1.1" 200 -
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 43276)
Traceback (most recent call last):
  File "/usr/lib/python2.7/SocketServer.py", line 593, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "widgets_overview_app.py", line 22, in __init__
    super(MyApp, self).__init__(*args)
  File "/usr/lib/python2.7/SocketServer.py", line 649, in __init__
    self.handle()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 328, in handle_one_request
    method()
  File "/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/server.py", line 540, in do_GET
    self.process_all(function)
  File "/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/server.py", line 574, in process_all
    with open(filename, 'r+b') as f:
IOError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/res/style.css'
----------------------------------------
127.0.0.1 - - [03/Nov/2015 21:25:15] "GET /res/logo.png HTTP/1.1" 200 -
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 43277)
Traceback (most recent call last):
  File "/usr/lib/python2.7/SocketServer.py", line 593, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "widgets_overview_app.py", line 22, in __init__
    super(MyApp, self).__init__(*args)
  File "/usr/lib/python2.7/SocketServer.py", line 649, in __init__
    self.handle()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 328, in handle_one_request
    method()
  File "/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/server.py", line 540, in do_GET
    self.process_all(function)
  File "/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/server.py", line 574, in process_all
    with open(filename, 'r+b') as f:
IOError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/res/logo.png'
----------------------------------------
127.0.0.1 - - [03/Nov/2015 21:25:15] "GET /res/logo.png HTTP/1.1" 200 -
----------------------------------------
Exception happened during processing of request from ('127.0.0.1', 43278)
Traceback (most recent call last):
  File "/usr/lib/python2.7/SocketServer.py", line 593, in process_request_thread
    self.finish_request(request, client_address)
  File "/usr/lib/python2.7/SocketServer.py", line 334, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "widgets_overview_app.py", line 22, in __init__
    super(MyApp, self).__init__(*args)
  File "/usr/lib/python2.7/SocketServer.py", line 649, in __init__
    self.handle()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 340, in handle
    self.handle_one_request()
  File "/usr/lib/python2.7/BaseHTTPServer.py", line 328, in handle_one_request
    method()
  File "/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/server.py", line 540, in do_GET
    self.process_all(function)
  File "/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/server.py", line 574, in process_all
    with open(filename, 'r+b') as f:
IOError: [Errno 13] Permission denied: '/usr/local/lib/python2.7/dist-packages/remi-0.1-py2.7.egg/remi/res/logo.png'
```

Am I doing anything wrong?

Also Chrome (in ubuntu 14.04) says "Connection is closed" when I run the script. And the website looks wrong.

**Comment by dddomodossola** [comment]
You should avoid to install with sudo, it is not required. Maybe the cause is it. Can you try to install the library without sudo? Right now I am not on a linux platform, I can only try tomorrow.

**Comment by awesomebytes** [comment]
Thank you for the lightning fast response. I'll try installing somewhere else and adding that to my PYTHONPATH (altho I dislike doing that as I always forget :P).

**Comment by awesomebytes** [comment]
I made a directory to store the lib and I'm still getting an error to install it:
```
sam@think:~/web_dynamic_reconf_ws/src/remi$ python setup.py install --prefix /home/sam/web_dynamic_reconf_ws/remi_lib
running install
running bdist_egg
running egg_info
writing remi.egg-info/PKG-INFO
error: [Errno 13] Permission denied: 'remi.egg-info/PKG-INFO'
```

I imagine you use a different setup, virtualenvs or something like that? But I've never used that.

**Comment by awesomebytes** [comment]
Now I just tried to do my first virtualenv for trying to install this, but it's still not working :/ Man I must be doing stuff very wrong.

```
(remi_env)sam@think:~/web_dynamic_reconf_ws/src/remi$ python setup.py install
running install
running bdist_egg
running egg_info
writing dependency_links to remi.egg-info/dependency_links.txt
error: [Errno 13] Permission denied: 'remi.egg-info/dependency_links.txt'
```

**Comment by awesomebytes** [comment]
Ok, this last error was generated because I used sudo previously and it wrote the remi.egg-info directory all with root permissions, so further runs of install when creating that temp folder didn't work!
My bad! Now it works :)

**Comment by dddomodossola** [comment]
;-) Happy to see you solved.

**Comment by awesomebytes** [comment]
Anyways if I want to use this in production I'll need to be able to install it somehow as a debian or pip install :P Are there plans for that?

**Comment by nzjrs** [comment]
You can pip install from the git url until a release is made.

**Comment by awesomebytes** [comment]
Didn't know the trick of pip installing from a git .zip url, nice, thanks!

**Comment by TylerDKnight** [comment]
Environment: I ran into this issue trying to set up a repo in a Juypter notebook on a Google Cloud VM. 

Problem: `setup.py build` and `setup.py install` were failing because of permissions issues. 

My specific output was:
`jupyter@tensorflow-2-vm:~/flownet2-pytorch/networks/resample2d_package$ python setup.py buildrunning build
running build_ext
building 'resample2d_cuda' extension
creating build/temp.linux-x86_64-3.7
error: could not create 'build/temp.linux-x86_64-3.7': Permission denied`

**Solution**: for me it was literally a permission issue. Doh. Somehow the blocked file belonged to `root`, not me the user, `jupyter`. To check if this is your problem too, `cd` to the directory of the offending file(s) and run `ls -l` to view the file permissions/owners. The file you are trying to run should belong to you and be fully executable. This is what it looked like for me while causing problems:
`rwxr-xr-x 2 root root 4096 Apr 22 07:33 temp.linux-x86_64-3.5` (note how it says `root root`)
this was after I fixed it:
`drwxr-xr-x 2 jupyter jupyter 4096 Apr 22 07:33 temp.linux-x86_64-3.5` (changed to my username)

To change the owner run `sudo chown -R $USER:$USER [FILE or FOLDER]`

Now in the appropriate directory, re-try with admin status `sudo python setup.py build`

Hope this helps someone out of a .pkl
